lib/cylc/scheduler.py

```diff
@@ -1199,60 +1015,50 @@ def configure_suite(self, reconfigure=False):
                 self.config.get_linearized_ancestors())
         else:
             # Things that can't change on suite reload.
-
-            self.state_dumper = SuiteStateDumper(
-                self.suite, self.run_mode, self.initial_point,
-                self.final_point)
-
             run_dir = GLOBAL_CFG.get_derived_host_item(
                 self.suite, 'suite run directory')
             pri_db_path = os.path.join(
-                run_dir, 'state', CylcSuiteDAO.DB_FILE_BASE_NAME)
+                run_dir, CylcSuiteDAO.PRI_DB_FILE_BASE_NAME)
             pub_db_path = os.path.join(
-                run_dir, CylcSuiteDAO.DB_FILE_BASE_NAME)
-            if self.is_restart:
-                if (os.path.exists(pub_db_path) and
-                        not os.path.exists(pri_db_path)):
-                    # Backwards compatibility code for restarting at move to
-                    # new db location should be deleted at database refactoring
-                    print('Copy "cylc.suite.db" to "state/cylc.suite.db"')
-                    copyfile(pub_db_path, pri_db_path)
-            else:
+                run_dir, CylcSuiteDAO.PUB_DB_FILE_BASE_NAME)
+            if not self.is_restart:
                 # Remove database created by previous runs
-                if os.path.isdir(pri_db_path):
-                    rmtree(pri_db_path)
-                else:
-                    try:
-                        os.unlink(pri_db_path)
-                    except OSError:
-                        pass
+                try:
+                    os.unlink(pri_db_path)
+                except OSError:
+                    # Just in case the path is a directory!
+                    rmtree(pri_db_path, ignore_errors=True)
             # Ensure that:
             # * public database is in sync with private database
             # * private database file is private
             self.pri_dao = CylcSuiteDAO(pri_db_path)
             os.chmod(pri_db_path, 0600)
             self.pub_dao = CylcSuiteDAO(pub_db_path, is_public=True)
             self._copy_pri_db_to_pub_db()
-
-            self.hold_suite_now = False
-            self._pool_hold_point = None
+            pub_db_path_symlink = os.path.join(
+                run_dir, CylcSuiteDAO.DB_FILE_BASE_NAME)
+            try:
+                source = os.readlink(pub_db_path_symlink)
+            except OSError:
+                source = None
+            if source != CylcSuiteDAO.PUB_DB_FILE_BASE_NAME:
+                try:
+                    os.unlink(pub_db_path_symlink)
+                except OSError:
+                    pass
+                os.symlink(
```

Comment:
I think we want to suite to fail here if fails to create the symbolic link here.